compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp

```diff
@@ -330,13 +332,9 @@ enumerateMatmulTileX86_64(TypeRange elementTypes, ExecutableTargetAttr target) {
 /// greater than the values.
 /// TODO(#16933): Remove `hostDefinedUpperBound` once we can propagate such
 /// information to host. For now, they are defined by host.
-static TileMxNxK
-chooseMatmulTile(ArrayRef<TileMxNxK> enumeratedTiles, int64_t matmulNarrowM,
-                 int64_t matmulNarrowN,
-                 ArrayRef<int64_t> hostDefinedUpperBound = {}) {
-  assert((hostDefinedUpperBound.empty() || hostDefinedUpperBound.size() >= 3) &&
-         "expected hostDefinedUpperBound is empty or has upper bound for {M, "
-         "N, K}");
+static TileMxNxK chooseMatmulTile(ArrayRef<TileMxNxK> enumeratedTiles,
+                                  int64_t matmulNarrowM, int64_t matmulNarrowN,
+                                  IntegerAttr hostDefinedUpperBound) {
```

Comment:
`hostDefinedUpperBound` was a poor name (not descriptive). Maybe standardize on `maxPadding`.

Reply:
updated